system/lib/pthread/library_pthread.c

```diff
@@ -237,6 +237,9 @@ int usleep(unsigned usec)
 	int is_main_thread = emscripten_is_main_runtime_thread();
 	double now = emscripten_get_now();
 	double target = now + usec * 1e-3;
+#ifdef __EMSCRIPTEN__
```

Comment:
Very good point, I do have a concern there. Updated this pull request to avoid FFI transition when profiling is not enabled. Can you take a peek at that, how does it look now?